### ISSUE TITLE
multinomial performance regressed 2x from 0.4.1 to 1.0.1

### ISSUE THREAD

**Issue by PetrochukM** [pytorch/pytorch#16939]
## 🐛 Bug

```
PyTorch version:  0.4.1
MKL:  True
Iterations:  48000
Threads:  1
Size:  256
torch.multinomial(a, 1): 0.318154849
```

```
PyTorch version:  1.0.1.post2
MKL:  True
Iterations:  48000
Threads:  1
Size:  256
torch.multinomial(a, 1): 0.640500445
```

## To Reproduce

```python3
import torch
from timeit import timeit

# Some defaults
number = 48000
n = 256
torch.set_grad_enabled(False)
print('PyTorch version: ', torch.__version__)
print('MKL: ', torch.backends.mkl.is_available())
print('Iterations: ', number)
print('Threads: ', torch.get_num_threads())
print('Size: ', n)

a = torch.rand(256)

# Test for performance
kwargs = {'globals': globals(), 'number': number}
print('torch.multinomial(a, 1):', timeit("torch.multinomial(a, 1)", **kwargs))
```

## Expected behavior

Expect same or better performance with torch 1.0.1.

## Environment

```
Collecting environment information...
PyTorch version: 1.0.1.post2
Is debug build: No
CUDA used to build PyTorch: None

OS: Mac OSX 10.14.1
GCC version: Could not collect
CMake version: version 3.12.2

Python version: 3.7
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip] numpy==1.15.4
[pip] numpydoc==0.8.0
[pip] pytorch-nlp==0.3.7.post1
[pip] torch==1.0.1.post2
[pip] torchvision==0.2.1
[conda] blas                      1.0                         mkl  
[conda] mkl                       2019.1                      144  
[conda] mkl-include               2019.1                      144  
[conda] mkl-service               1.1.2            py37hfbe908c_5  
[conda] mkl_fft                   1.0.6            py37h27c97d8_0  
[conda] mkl_random                1.0.2            py37h27c97d8_0  
[conda] pytorch                   1.0.1                   py3.7_2    pytorch
[conda] pytorch-nlp               0.3.7.post1              pypi_0    pypi
[conda] torchvision               0.2.1                      py_2    pytorch
```

**Comment by ifedan** [comment]
I will investigate this

**Comment by vishwakftw** [comment]
I believe this can be closed now.

**Comment by soumith** [comment]
closed via #17121
